include/at_chops/aes_ctr.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#ifdef BUILD_MBEDTLS
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include <string.h>
+
+#define IV_AMOUNT_BYTES 16
+
+typedef enum {
+    AES_128 = 128,
+    AES_192 = 192,
+    AES_256 = 256,
+} AESKeySize;
+
+typedef struct {
+    size_t reslen; // length of the result written
+    unsigned char *res; // result of the encryption/decryption
+    int status; // status code of the operation
```

Comment:
For the same reason, you probably want status as first not last, so that status is at 0, reslen is at sizeof(int). Instead of reslen at 0, status at reslen * sizeof(unsigned char) which is added complexity

Reply:
https://github.com/atsign-foundation/at_c/pull/7/commits/fe41978068847ca1fb850acf39b189df297aa9e3